src/tutorial/microbe_stage/DayNightTutorial.cs

```diff
@@ -21,8 +21,11 @@ public override bool CheckEvent(TutorialState overallState, TutorialEventType ev
         {
             case TutorialEventType.MicrobePlayerEnterSunlightPatch:
             {
+                MicrobeHUD hud = ((HUDEventArgs)args).HUD;
```

Comment:
I think it is pretty dirty to pass the HUD like this.

Also the environment panel has to be shown for temperature change with thermosynthesis.

I don't immediately have a good idea here, but I think that some kind of system like "wants pause" going through the tutorial state would be better. Maybe some kind of event system where the TutorialState offers an event callback any active HUD registers to, and individual tutorials can then send events through like "EnvironmentPanelRequired"?

Reply:
I agree that the args is a little abused in our tutorials. If you have a better pattern you prefer, however, you'll probably need to explain to me how it's going to work...

Reply:
I did lay out a rough idea. After having it stew in my head for a while, I think it is probably the best option, so adding a method in TutorialState to trigger a tutorial event that is listened to by active HUD instances, would be the way to go in my opinion. Then the tutorials can send events like "compounds panel should be visible" etc. and the relevant HUD then reacts to it.